src/vellum/workflows/types/utils.py

```diff
@@ -150,14 +150,30 @@ def deepcopy_with_exclusions(
 ) -> _T:
     cls = obj.__class__
     new_instance = cls.__new__(cls)
-    new_instance.__dict__.update(obj.__dict__)
 
     exclusions = exclusions or {}
 
-    for key, value in obj.__dict__.items():
-        if key in exclusions:
-            continue
-        new_instance.__dict__[key] = deepcopy(value, memo)
+    # Check if object has a lock for thread-safe copying
+    obj_lock = getattr(obj, "__lock__", None)
+    if obj_lock is not None:
```

Comment:
would love to start with the `keys()` change mentioned above to see if it helps. This logic is all try/except'd so low urgency on more sweeping changes here involving lock's